src/lib.rs

```diff
@@ -285,6 +288,25 @@ impl WalkDir {
         self.opts.max_open = n;
         self
     }
+
+    /// Set a compare function.
+    /// If a compare function is set, WalkDir will return all paths in sorted
+    /// order.
+    ///
+    /// ```rust,no-run
+    /// use std::cmp;
+    /// use std::ffi::OsString;
+    /// use walkdir::WalkDir;
+    ///
+    /// fn string_compare(a: &OsString, b: &OsString) -> cmp::Ordering {
+    ///     a.cmp(b)
+    /// }
+    /// WalkDir::new("foo").sort(Some(string_compare));
+    /// ```
+    pub fn sort(mut self, sorter: Option<fn(&OsString,&OsString) -> std::cmp::Ordering>) -> Self {
```

Comment:
Yes, I expressly used a plain function in the first version just to get feedback.

I also thought that passing a parameter to new() would be ugly and avoided going all out with a Trait. Using a trait would be simplest for the user I think.